### PR TITLE
Treat captions as tracks in standard workflows

### PR DESCRIPTION
**Pull request by Arnei**
*This PR aims at contributing to the implementation of #4407, but touches on just one of the aspects to realize #4407's vision. It should therefore likely not be merged alone, but alongside other PRs.*

Instead of treating captions as something special, this commit attempts to change the workflows to treat them as tracks, subjecting them to the same operations.
Actually has to skip some operations in partial-publish that will cause an exception if given something that is not a media file, but that should be fine since those operations are irrelevant to subtitles.

### CODE REVIEW

**Comment by KatrinIhler** [comment]
Changes look reasonable. Will report once I have tested this a bit.

**Comment by KatrinIhler** [comment]
A couple of problems I noticed during testing:

1. The explicit snapshotting of captions/* should also be removed from publish.xml  and partial-error.xml.
2. Publication of captions doesn't work for a republish because captions/prepared isn't archived the first time because it is changed (without copy) to presenter/work, and will not created during the second run because presenter and/or presentation/prepared _was_ archived. 
For this to work properly, the easiest way is to use the copy option when changing captions/prepared to captions/work (although the whole */prepared stuff isn't terribly useful right now).

**Review by KatrinIhler — CHANGES_REQUESTED** [review]
A couple of problems I noticed during testing:

1. The explicit snapshotting of captions/* should also be removed from publish.xml and partial-error.xml.
2.  Publication of captions doesn't work for a republish because captions/prepared isn't archived the first time because it is changed (without copy) to presenter/work, and will not created during the second run because presenter and/or presentation/prepared was archived.
    For this to work properly, the easiest way is to use the copy option when changing captions/prepared to captions/work (although the whole */prepared stuff isn't terribly useful right now).

**Comment by mliradelc** [comment]
@KatrinIhler I don't have strong opinions with the standard workflows (We use custom ones) Are the latest changes OK with you?

File: etc/workflows/fast.yaml [hunk 1]
```diff
@@ -44,6 +44,16 @@ operations:
...
 
+  - id: tag
+    fail-on-error: true
+    exception-handler-workflow: "partial-error"
+    description: "Tag captions for publication"
+    configurations:
+      - source-flavor: "captions/source"
+      - target-flavor: "captions/preview"
+      - target-tags: "+engage-download"
+      - copy: true
```
Comment (mliradelc) [inline etc/workflows/fast.yaml:55]: Should you add?
`generator:unknown`
`generator-type:manual`
`lang:unknown`

**Review by mliradelc — APPROVED** [review]
For me is ok, only one question regarding the tags.

File: etc/workflows/fast.yaml (hunk 1 above)
Comment (Arnei) [reply etc/workflows/fast.yaml:55]: It is not necessary per se, tags are not variables that need to be initalized or anything like that. 

I don't think the workflows generally should make general assumptions about subtitle tags (was the caption written manually or automatically, the workflow does not know). Although since this is a workflow used primarily for testing, maybe the tags are nice for "demonstrating" how tags for captions are used in Opencast?

Bottom line, for the fast workflow in particular I don't really care. If you think it's a good idea, let's do it.

Comment (mliradelc) [reply etc/workflows/fast.yaml:55]: Fair enough. I'm only thinking in your changes in Paella, that can detect those tags for the subtitle selection.
